app/src/main/java/com/fcaputo/parkingapp/mvp/contract/ReservationContract.kt

```diff
@@ -0,0 +1,44 @@
+package com.fcaputo.parkingapp.mvp.contract
+
+import com.fcaputo.parkingapp.entity.Reservation
+import com.fcaputo.parkingapp.utils.DateTimePicker
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.fcaputo.parkingapp.utils.validation.ValidationResult
+import java.util.Calendar
+
+interface ReservationContract {
+    interface Model {
+        fun getParkingLotSize(): Int
+        fun storeReservation(reservation: Reservation)
+        fun getReservations(): List<Reservation>
+        fun validateData(startDate: Calendar, endDate: Calendar, parkingSpot: Int): ValidationResult
+    }
+
+    interface Presenter {
+        fun onDateTimeClick(picker: DateTimePicker)
+        fun onParkingSpotFocus()
+        fun onSaveButton()
+        fun onSettingsButton()
+    }
+
+    interface View {
+        fun showParkingLotSize(size: String)
+        fun onSettingsButton(onClick: () -> Unit)
+        fun navigateToSettings()
+        fun onSaveButton(onClick: () -> Unit)
+        fun onParkingSpotInputFocus(onFocus: () -> Unit)
+        fun onDateTimeInputPressed(onInputClick: (DateTimePicker) -> Unit)
+        fun showDatePicker(selectedDate: Calendar, pickerType: DateTimePicker)
+        fun showTimePicker(selectedTime: Calendar, pickerType: DateTimePicker)
+        fun getEditTextsContents(): List<CharSequence?>
```

Comment:
Do you really need the CharSequence to be nullable?

Reply:
No it really isn't. But I thought that since the editText `text` property returns an `Editable?`, it would be needed to consider its nullability

Reply:
well yeah, but you can capture that nullability and transform it to a non nullable, like 
If(Editable != null) {
  **add to list
} else {
  **add to list as EMPTY_STRING
} 
*theres a shorter way of doing this, but just to show the idea

Reply:
be careful, because in the view, you should not have an if statement

Reply:
yes, my bad. then you can make a .orEmpty() to that Editable, try to avoid the nullables

Reply:
Finally I decided to use .toString() on each `Editable?` object. Documentation states that, in case of being null, .toString() returns the literal `"null"`
Then I check in the presenter for those conditions